helix-core/src/main/java/org/apache/helix/model/ClusterConfig.java

```diff
@@ -1255,4 +1257,22 @@ public void setPreferredScoringKeys(List<String> preferredScoringKeys) {
     _record.setListField(ClusterConfigProperty.PREFERRED_SCORING_KEYS.name(),
         preferredScoringKeys);
   }
+
+  public boolean isParticipantDeregistrationEnabled() {
+    return _record.getBooleanField(ClusterConfigProperty.PARTICIPANT_DEREGISTRATION_ENABLED.name(),
+        false);
+  }
+
+  public void setParticipantDeregistrationEnabled(boolean enabled) {
+    _record.setBooleanField(ClusterConfigProperty.PARTICIPANT_DEREGISTRATION_ENABLED.name(), enabled);
+  }
+
+  public long getParticipantDeregistrationTimeout() {
+    return _record.getLongField(ClusterConfigProperty.PARTICIPANT_DEREGISTRATION_TIMEOUT.name(),
+        -1);
+  }
+
+  public void setParticipantDeregistrationTimeout(long timeout) {
+    _record.setLongField(ClusterConfigProperty.PARTICIPANT_DEREGISTRATION_TIMEOUT.name(), timeout);
+  }
```

Comment:
Let's simplify this? If the timeout is not setup or -1, the feature is not enabled? Other cluster config is resource level, so some of resources may not have the configurations. But this is instance level. I dont think we need a switcher to control the feature.

Reply:
Adjusted to only rely on the timeout value. If timeout is -1 or lower, then it is turned off. If there is no timeout set, the default value is -1 which is interpreted as disabled